src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int RESIZE_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75F;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (table.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndexFor(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> tempNode = table[index];
+            while (tempNode != null) {
+                if (Objects.equals(tempNode.key, key)) {
+                    tempNode.value = value;
+                    return;
+                }
+                if (tempNode.next == null) {
+                    tempNode.next = newNode;
+                    size++;
+                    return;
+                }
+                tempNode = tempNode.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndexFor(key);
+        Node<K, V> tempNode = table[index];
+        while (tempNode != null) {
+            if (Objects.equals(tempNode.key, key)) {
+                return tempNode.value;
+            }
+            tempNode = tempNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndexFor(Object key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        if (threshold == size) {
+            size = 0;
+            int newSize = table.length * RESIZE_FACTOR;
+            threshold = (int) (newSize * LOAD_FACTOR);
+            transferToNewTable(newSize);
+        }
+    }
+
+    private void transferToNewTable(int newSize) {
```

Comment:
let's combine methods transferToNewTable() and resize(), because they do the same logic